cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -1413,19 +1221,49 @@ public WorkflowToken getWorkflowToken(ProgramId workflowId, String workflowRunId
     Set<RunId> runIds = new HashSet<>();
     Iterables.addAll(runIds, getRunningInRangeForStatus(TYPE_RUN_RECORD_COMPLETED, startTimeInSecs, endTimeInSecs,
                                                         scanTimeoutMills));
-    Iterables.addAll(runIds, getRunningInRangeForStatus(TYPE_RUN_RECORD_SUSPENDED, startTimeInSecs, endTimeInSecs,
-                                                        scanTimeoutMills));
-    Iterables.addAll(runIds, getRunningInRangeForStatus(TYPE_RUN_RECORD_STARTED, startTimeInSecs, endTimeInSecs,
-                                                        scanTimeoutMills));
-    Iterables.addAll(runIds, getRunningInRangeForStatus(TYPE_RUN_RECORD_STARTING, startTimeInSecs, endTimeInSecs,
+    Iterables.addAll(runIds, getRunningInRangeForStatus(TYPE_RUN_RECORD_ACTIVE, startTimeInSecs, endTimeInSecs,
                                                         scanTimeoutMills));
     return runIds;
   }
 
   /**
-   * @return true if the row key is value is greater or than or equal to the expected version
+   * Get the run count of the given program.
+   *
+   * @param programId the program to get the count
+   * @return the number of run count
+   */
+  public int getProgramRunCount(ProgramId programId) {
+    MDSKey key = getProgramKeyBuilder(TYPE_COUNT, programId).build();
+    byte[] count = getValue(key);
+    return count == null ? 0 : (int) Bytes.toLong(count);
+  }
+
+  /**
+   * Get the run counts of the given program collections.
+   *
+   * @param programIds the collection of program ids to get the program
+   * @return the map of the program id to its run count
+   */
+  public Map<ProgramId, Integer> getProgramRunCounts(Collection<ProgramId> programIds) {
+    Map<MDSKey, ProgramId> mdsKeyProgramIdMap = new HashMap<>();
+    for (ProgramId programId : programIds) {
+      MDSKey key = getProgramKeyBuilder(TYPE_COUNT, programId).build();
+      mdsKeyProgramIdMap.put(key, programId);
+    }
+    Map<MDSKey, byte[]> counts = getKV(mdsKeyProgramIdMap.keySet());
+    Map<ProgramId, Integer> result = new LinkedHashMap<>();
+    for (MDSKey mdsKey : mdsKeyProgramIdMap.keySet()) {
+      byte[] count = counts.getOrDefault(mdsKey, Bytes.toBytes(0L));
```

Comment:
Don't do getOrDefault with a new `byte[]` every time, as you only need the `0` if it is missing.